### ISSUE TITLE
Why was ExpectedExceptionAttribute removed?

### ISSUE THREAD

**Issue by pjmagee** [nunit/nunit#799]
I don't see a great benefit from removing this functionality in the next version of NUnit. Now for what used to look like clear simple methods, now have to be wrapped inside an Action

```c#

Assert.Throws<Exception>(() => 
{
        // Test Code
});

```

When before if you had a lot of simple tests the `[ExpectedExceptionAttribute]` was enough.

**Comment by rprouse** [comment]
`ExpectedException` was removed because it encourages bad practices and it was generally felt that removing it helped developers fall into the [pit of success](http://english.stackexchange.com/questions/77535/what-does-falling-into-the-pit-of-success-mean).

The `ExpectedException` attribute is only really safe if there is one line of code in your test method or the other lines cannot possibly throw the given exception. `ExpectedException` only checks that one of the lines of code in a method threw the exception and if another line throws the exception you are expecting it could easily mask errors. In contrast, `Assert.Throw` requires a bit more code, but it is very explicit in which line or group of code you are expecting to throw an exception.

We realize that migrating older code that made extensive use of `ExpectedException` might be a lot of work, so we wrote a simplified version of [ExpectedException](https://github.com/nunit/nunit-csharp-samples/blob/master/ExpectedExceptionExample/ExpectedExceptionAttribute.cs) as an extension example. Just copy the class into your code and anything but advanced use of the old `ExpectedException` class will work.

Migrating to NUnit 3.0 might cause a small amount of short term pain because of changes like this, but we did not take them lightly. They were only done after extensive consultation with the community and we believe that they are beneficial in the long run.
